torch/distributed/distributed_c10d.py

```diff
@@ -929,6 +940,7 @@ def _new_process_group_helper(
     pg_options=None,
     group_name=None,
     timeout=default_pg_timeout,
+    pg_tag=None
```

Comment:
any difference between `pg_tag` and `group_name` arguments? I think group_name is deprecated but there is still an existing map and argument for it, we might want to just reuse that?

Reply:
It's a bit different, reusing would require us first to stop the current usage.
We should do it, but who knows how much breakage it might cause.